teku/src/integration-test/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommandTest.java

```diff
@@ -0,0 +1,158 @@
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.junit.jupiter.MockServerExtension;
+import org.mockserver.model.JsonBody;
+import org.mockserver.model.Parameter;
+import tech.pegasys.teku.api.ConfigProvider;
+import tech.pegasys.teku.bls.BLSTestUtil;
+import tech.pegasys.teku.cli.BeaconNodeCommand;
+import tech.pegasys.teku.infrastructure.logging.LoggingConfigurator;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+
+@ExtendWith(MockServerExtension.class)
+public class VoluntaryExitCommandTest {
+
+  private final StringWriter stringWriter = new StringWriter();
+  private final PrintWriter outputWriter = new PrintWriter(stringWriter, true);
+  private final PrintWriter errorWriter = new PrintWriter(stringWriter, true);
+  private final LoggingConfigurator loggingConfigurator = mock(LoggingConfigurator.class);
+
+  private final BeaconNodeCommand.StartAction startAction =
+      mock(BeaconNodeCommand.StartAction.class);
+
+  private final Spec testSpec =
+      SpecFactory.create(
+          Resources.getResource("tech/pegasys/teku/cli/subcommand/test-spec.yaml").getPath());
+
+  private ClientAndServer mockBeaconServer;
+
+  private final BeaconNodeCommand beaconNodeCommand =
+      new BeaconNodeCommand(
+          outputWriter, errorWriter, Collections.emptyMap(), startAction, loggingConfigurator);
+
+  private final ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+
+  private final String pubKey =
+      "0x82c2a92c6823d43bf11215eddd0f706f168945b298bf718288cf78fcef81df19cd1e5c9c49e7b8d689722fb409f0c0a4";
+
+  @BeforeEach
+  public void setup(ClientAndServer server) throws IOException {
+    this.mockBeaconServer = server;
+    configureSuccessfulSpecResponse(mockBeaconServer);
+    configureSuccessfulHeadResponse(mockBeaconServer);
+    configureSuccessfulGenesisResponse(mockBeaconServer);
+    configureSuccessfulValidatorResponse(mockBeaconServer);
+    configureSuccessfulVoluntaryExitResponse(mockBeaconServer);
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @AfterEach
+  public void tearDown() {
+    mockBeaconServer.reset();
+    System.setOut(System.out);
+  }
+
+  @Test
+  public void shouldExitValidatorWithPubKey() {
+    final String[] argsNetworkOptOnParent =
+        new String[] {
+          "voluntary-exit",
+          "--data-validator-path",
+          Resources.getResource("tech/pegasys/teku/cli/subcommand/validator").getPath(),
+          "--beacon-node-api-endpoint",
+          getMockBeaconServerEndpoint(mockBeaconServer),
+          "--confirmation-enabled",
+          "false",
+          "--validator-public-keys",
+          pubKey + "," + BLSTestUtil.randomPublicKey(17).toString()
+        };
+    int parseResult = beaconNodeCommand.parse(argsNetworkOptOnParent);
+    assertThat(parseResult).isEqualTo(0);
+    assertThat(stdOut.toString(UTF_8)).contains("Exit for validator 82c2a92 submitted.");
```

Comment:
I think we could have only 2 keys instead of 10 in the `key-manager` dir.
It could be good, instead, to have a key in the standard key management specifying `--validator-keys` to check the interaction with this other additional key set.

and for completeness we can assert that the stdOut doesn't contain the Exit output for the key that wasn't specified in the argument but present in the key store. Just to be sure that we are not exiting other keys.

Reply:
I agree but the problem is that the default data path points to the machine FS.
I will reduce the keys though

Reply:
I will reduce the keys though